modin/core/storage_formats/base/query_compiler.py

```diff
@@ -89,7 +89,7 @@ def axis_setter(self, labels):
 # Currently actual arguments are placed in the methods docstrings, but since they're
 # not presented in the function's signature it makes linter to raise `PR02: unknown parameters`
 # warning. For now, they're silenced by using `noqa` (Modin issue #3108).
-class BaseQueryCompiler(abc.ABC):
```

Comment:
Just unify the approach.

Reply:
`BaseQueryCompiler` should be split up into `BaseQueryCompiler`, which has abs methods, and `Default(2)PandasQueryCompiler`, which has default to pandas behavior (aka BaseOnPython execution, which should probably be renamed too).